code/modules/mob/living/living.dm

```diff
@@ -650,38 +650,38 @@
 /mob/living/proc/makeTrail(turf/T)
 	if(!has_gravity(src))
 		return
-	var/blood_exists = 0
-
-	for(var/obj/effect/decal/cleanable/trail_holder/C in loc) //checks for blood splatter already on the floor
-		blood_exists = 1
-	if(isturf(loc))
-		var/trail_type = getTrail()
-		if(trail_type)
-			var/brute_ratio = round(getBruteLoss()/maxHealth, 0.1)
-			if(blood_volume && blood_volume > max(BLOOD_VOLUME_NORMAL*(1 - brute_ratio * 0.25), 0))//don't leave trail if blood volume below a threshold
-				blood_volume = max(blood_volume - max(1, brute_ratio * 2), 0) 					//that depends on our brute damage.
-				var/newdir = get_dir(T, loc)
-				if(newdir != src.dir)
-					newdir = newdir | dir
-					if(newdir == 3) //N + S
-						newdir = NORTH
-					else if(newdir == 12) //E + W
-						newdir = EAST
-				if((newdir in GLOB.cardinal) && (prob(50)))
-					newdir = turn(get_dir(T, loc), 180)
-				if(!blood_exists)
-					new /obj/effect/decal/cleanable/trail_holder(loc)
-				for(var/obj/effect/decal/cleanable/trail_holder/TH in loc)
-					if((!(newdir in TH.existing_dirs) || trail_type == "trails_1" || trail_type == "trails_2") && TH.existing_dirs.len <= 16) //maximum amount of overlays is 16 (all light & heavy directions filled)
-						TH.existing_dirs += newdir
-						TH.overlays.Add(image('icons/effects/blood.dmi', trail_type, dir = newdir))
-						TH.transfer_mob_blood_dna(src)
-						if(ishuman(src))
-							var/mob/living/carbon/human/H = src
-							if(H.dna.species.blood_color)
-								TH.color = H.dna.species.blood_color
-						else
-							TH.color = "#A10808"
+	if(!isturf(loc))
+		return
+	var/trail_type = getTrail()
+	if(!trail_type)
+		return
+	var/brute_ratio = round(getBruteLoss()/maxHealth, 0.1)
+	if(!blood_volume && !(blood_volume > max(BLOOD_VOLUME_NORMAL*(1 - brute_ratio * 0.25), 0)))// Okay let's dive into the maths. For every 50 brute damage taken, the minimal blood level you can have decreases by 12,5%
+		return
+	blood_volume = max(blood_volume - max(1, brute_ratio * 2), 0) 					   // The amount of blood lost per tile of movement is always at least 1cl, and every 50 damage after reaching 50 brute damage taken will increase the bleed by 1cl per tile
+	var/newdir = get_dir(T, loc)
+	if(newdir != dir)
+		newdir = newdir | dir
+		if(newdir == 3) //N + S
+			newdir = NORTH
+		else if(newdir == 12) //E + W
+			newdir = EAST
+	if(IS_DIR_CARDINAL(newdir) && (prob(50)))
+		newdir = turn(get_dir(T, loc), 180)
+	var/blood_exists = locate(/obj/effect/decal/cleanable/trail_holder) //checks for blood splatter already on the floor
```

Comment:
`locate` without `in` tries to fiend first occurance of the object of this type in the world, so you should specify WHERE to look for the type. (maybe I am wrong in this particular case, and there locate somehow searches the current loc, but I doubt it, the reference doesn't say anything about it)

Reply:
This should be better now? I really haven't used locate() before, so sorry if I used it wrong again

Reply:
Yep. Now it should work good.